### DISCUSSION TITLE
Ability to block Tesla car updates while tethered.

### DISCUSSION THREAD

**Discussion by gltovar** [tesla-android/issue-tracker#318] [Ideas]
Would it be possible to override Tesla car update downloads by ‘hijacking’ the url that tesla requests to check for updates? I issue I have is that the updates are massive and it chews through mobile data. Ideally Tesla themselves would add an option to prevent ‘automatic’ update downloads in their software, just notify users it is available.  But until then can we work around this?

**Comment by TastePurple** [comment]
This feature is already implemented in TA settings. Click on 'Network' and scroll down to 'Tesla Software Updates' and disable it.

**Reply by atarev-crypto** [reply]
Can you share how to do this (how this is implemented, not how to turn it on in TA)? Guessing a DNS block or similar? I'd like to do the same when using mobile data from my phone.

**Comment by AndrewBucklin** [comment]
How about the ability to block ALL traffic from the car? I only want Tesla Android to have internet access. Basically looking for a way to allow the car to only talk to Tesla Android and nothing else. Perhaps the built-in DHCP server can be set to never provide a default gateway IP?

**Comment by mikegapinski** [comment]
@AndrewBucklin I already have a solution like this implemented but it only handles certain endpoints. I can easily implement what you said (0 internet to the car, only TA is online). Struggling to find a use case for this though

**Reply by AndrewBucklin** [reply]
Some of us have _reasons_ or personal beliefs for never wanting the car to phone home and some have even gone as far as removing the LTE modem from the car (thus rendering the built-in navigation unusable, which is a major reason for the installation of Tesla Android).

**Reply by mikegapinski** [reply]
makes perfect sense, this would be some kind of a stealth mode. If you can give me an iptables rule for that I can get it tested tomorrow. The car would only need to access 104.248.101.213(spoofed address that hosts the frontend for TA and 172.16.0.1)

**Reply by thorwaler** [reply]
Hi Mike, 
did you in the end manage to do this? I would also love my Tesla not to communicate all the data with the mothership.
